File: TP01/Q01/palindromo.java
package TP01.Q01;

import TP01.MyIO;

public class palindromo {

    public static void main(String[] args) {

        String palavra = MyIO.readLine();
        while(!palavra.equalsIgnoreCase("FIM")){
            if(isPalindromo(palavra)){
                System.out.println("SIM");
            }else{
                System.out.println("NAO");
            }
            palavra = MyIO.readLine();
        }

    }

    public static boolean isPalindromo(String palavra){

        int i;
        int j;
        for(i = 0, j = (palavra.length()-1); i < palavra.length() / 2; i++, j--){
            if(palavra.charAt(i) != palavra.charAt(j)) {
                return false;
            }
        }
        return true;
    }
}

File: TP01/TP01Q02/palindromo.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

bool ehPalindromo(char palavra[]){
    
    bool result = true;
    int tamanho = strlen(palavra);
    for(int i = 0, j = tamanho - 1; i < tamanho/2; i++, j--){
        if(palavra[i] != palavra[j]){
            result = false;
            i = tamanho;
        }
    }
    return result;
}

int main(){
 
    char* palavra = (char*)malloc(200 * sizeof(char));
    bool palindromo = false;
    scanf(" %[^\r\n%*c]", palavra);
    while(strcmp(palavra,"FIM") != 0){
        palindromo = ehPalindromo(palavra);
        if(palindromo){
            printf("SIM\n");
        }else{
            printf("NAO\n");
        }
        scanf(" %[^\r\n]%*c", palavra);
    }
    return 0;
}